lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -3402,6 +3402,75 @@ class DecomposeAtenVarMeanOp : public OpRewritePattern<AtenVarMeanOp> {
 };
 } // namespace
 
+namespace {
+class DecomposeAtenMovedimIntOp : public OpRewritePattern<AtenMovedimIntOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMovedimIntOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.getSelf();
+    std::optional<unsigned> maybeInputRank = getTensorRank(input);
+    if (!maybeInputRank) {
+      return rewriter.notifyMatchFailure(
+          op, "expected input tensor to have a rank");
+    }
+    unsigned inputRank = *maybeInputRank;
+    Value result = input;
+    if (inputRank != 0) {
+      int64_t srcDimInt, dstDimInt;
+      if (matchPattern(op.getSource(), m_TorchConstantInt(&srcDimInt))) {
+        srcDimInt = toPositiveDim(srcDimInt, inputRank);
+        if (!isValidDim(srcDimInt, inputRank))
+          return rewriter.notifyMatchFailure(op, "source is not a valid dim");
+      } else {
+        return rewriter.notifyMatchFailure(op, "source is not a constant int");
+      }
+      if (matchPattern(op.getDestination(), m_TorchConstantInt(&dstDimInt))) {
+        dstDimInt = toPositiveDim(dstDimInt, inputRank);
+        if (!isValidDim(dstDimInt, inputRank))
+          return rewriter.notifyMatchFailure(op,
+                                             "destination is not a valid dim");
+      } else {
+        return rewriter.notifyMatchFailure(op,
+                                           "destination is not a constant int");
+      }
+      SmallVector<int64_t> srcDims, dstDims, dimsOrder;
+      for (unsigned i = 0; i < inputRank; i++) {
+        srcDims.push_back(i);
+        dstDims.push_back(i);
+        dimsOrder.push_back(-1);
+      }
+      dimsOrder[dstDimInt] = srcDimInt;
+      srcDims[srcDimInt] = -1;
+      dstDims[dstDimInt] = -1;
+      SmallVector<int64_t> sourceDims, destinationDims;
+      for (unsigned i = 0; i < inputRank; i++) {
+        if (srcDims[i] != -1)
+          sourceDims.push_back(srcDims[i]);
+        if (dstDims[i] != -1)
+          destinationDims.push_back(dstDims[i]);
+      }
+      unsigned restDim = inputRank - 1;
+      for (unsigned i = 0; i < restDim; i++)
+        dimsOrder[destinationDims[i]] = sourceDims[i];
```

Comment:
I think you can do instead

```c++
  SmallVector<int> dimsOrder(llvm::iota_range<int>(0, inputRank, /*inclusive=*/false));
  dimsOrder.erase(dimsOrder.begin() + srcDimInt);
  dimsOrder.insert(dimsOrder.begin() + dstDimInt, srcDimInt);
```

Reply:
The algo is taken from here: https://github.com/pytorch/pytorch/blob/c47bdd7522c7146c091c95ff5b4ce354d7a8870c/aten/src/ATen/native/TensorShape.cpp#L3660. I don't understand how these 3 lines of code would work in place of this algo.